internal/fs/fs_reader.go

```diff
@@ -103,17 +103,32 @@ func (fs *Reader) Stat(name string) (os.FileInfo, error) {
 // describes the symbolic link.  Lstat makes no attempt to follow the link.
 // If there is an error, it will be of type *PathError.
 func (fs *Reader) Lstat(name string) (os.FileInfo, error) {
-	switch name {
-	case fs.Name:
-		return fs.fi(), nil
-	case "/", ".":
+	getDirInfo := func(name string) os.FileInfo {
 		fi := fakeFileInfo{
 			name:    name,
```

Comment:
This needs to be `path.Base(name)`:
```suggestion
			name:    path.Base(name),
```